lib/fake_ifdh.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python3 -I
+
+# fake_ifdh -- get rid of ifdhc dependency by providing a few
+#              bits of ifdh behavior
+#
+# COPYRIGHT 2021 FERMI NATIONAL ACCELERATOR LABORATORY
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import os
+import time
+import argparse
+
+VAULT_HOST = "fermicloud543.fnal.gov"
+DEFAULT_ROLE = "Analysis"
+
+
+def getTmp():
+    return os.environ.get("TMPDIR", "/tmp")
+
+
+def getExp():
+    for ev in ["GROUP", "EXPERIMENT", "SAM_EXPERIMENT"]:
+        if os.environ.get(ev, None):
+            return os.environ.get(ev)
+    # otherwise guess primary group...
+    f = os.popen("id -gn", "r")
+    exp = f.read()
+    f.close()
+    return exp
+
+
+def getRole(role_override=None):
+    if role_override:
+        return role_override
+    elif os.environ["USER"][-3:] == "pro":
+        return "Production"
+    else:
+        return DEFAULT_ROLE 
+
+
+def checkToken(tokenfile):
+    f = os.popen("decode_token.sh -e exp %s 2>/dev/null" % tokenfile, "r")
```

Comment:
What happens if this fails? Can we check for any errors here and handle them?

Reply:
If this fails we have an empty expiration string and the check fails.

Reply:
I mean in terms of any exceptions or errors running the actual `decode_token.sh` script.  If there is an exception running it, I assume we want to exit at that point (and print the error to screen), right?  If so, I'm fine leaving this as you have it.

Reply:
I don't think os.popen raises anything (as long as you pass it a string); you have to check the returned handle, and the  value from the close method on the handle, to see if anything went wrong.